### PR TITLE
Support MONOTONIC, PROCESS_CPUTIME_ID, THREAD_CPUTIME_ID clocks

### PR DESCRIPTION
**Pull request by anakrish**
Signed-off-by: Anand Krishnamoorthi <anakrish@microsoft.com>

### CODE REVIEW

**Comment by anakrish** [comment]
bors try

File: enclave/core/time.c [hunk 1]
```diff
@@ -1,9 +1,12 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
+#include "openenclave/bits/time.h"
 #include <openenclave/bits/types.h>
 #include <openenclave/internal/calls.h>
+#include <openenclave/internal/raise.h>
 #include <openenclave/internal/time.h>
+#include "openenclave/bits/result.h"
```
Comment (dthaler) [inline enclave/core/time.c:9]: why is this one out of alphabetical order?
Also lines 4 and 9 use quotes whereas the other lines use <> which seems inconsistent

File: host/linux/time.c [hunk 2]
```diff
@@ -1,11 +1,13 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
+#include "openenclave/bits/time.h"
```
Comment (dthaler) [inline host/linux/time.c:4]: move down into alphabetical order and make all OE headers be consistent (I'd prefer all "" not <>)

File: host/linux/time.c [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 #include <errno.h>
 #include <openenclave/internal/localtime.h>
 #include <openenclave/internal/time.h>
 #include <time.h>
 #include "../ocalls/ocalls.h"
+#include "openenclave/bits/types.h"
```
Comment (dthaler) [inline host/linux/time.c:10]: ditto

File: host/windows/time.c [hunk 4]
```diff
@@ -74,3 +74,78 @@ int oe_localtime(const time_t* timep, struct tm* result)
...
+                    GetCurrentProcess(),
+                    &creation_time,
+                    &exit_time,
+                    &kernel_time,
+                    &user_time) == 0)
+            {
+                errno = EINVAL;
+                uint64_t total_time = kernel_time.u64 + user_time.u64;
+                tp->tv_sec = total_time / 10000000;
+                tp->tv_nsec = (total_time % 10000000) * 100;
```
Comment (dthaler) [inline host/windows/time.c:119]: Shouldn't these be set in the success case instead of the error case?

File: host/windows/time.c [hunk 5]
```diff
@@ -74,3 +74,78 @@ int oe_localtime(const time_t* timep, struct tm* result)
...
+                    GetCurrentThread(),
+                    &creation_time,
+                    &exit_time,
+                    &kernel_time,
+                    &user_time) == 0)
+            {
+                errno = EINVAL;
+                uint64_t total_time = kernel_time.u64 + user_time.u64;
+                tp->tv_sec = total_time / 10000000;
+                tp->tv_nsec = (total_time % 10000000) * 100;
```
Comment (dthaler) [inline host/windows/time.c:137]: Shouldn't these be set in the success case instead of the error case?

File: include/openenclave/internal/time.h [hunk 6]
```diff
@@ -36,6 +37,20 @@ OE_EXTERNC_BEGIN
 
 uint64_t oe_get_time(void);
 
+/*
+ * Forward declaration of ocall since libc/CMakeLists.txt does not process any
+ * EDLs. The code is structured such that if this declaration does not match
+ * oeedger8r generated declaration, there will be compile error.
```
Comment (dthaler) [inline include/openenclave/internal/time.h:43]:
```suggestion
 * an oeedger8r generated declaration, there will be a compile error.
```
grammar nit

File: syscall/syscall.c [hunk 7]
```diff
@@ -26,6 +26,7 @@
 #include <openenclave/internal/syscall/sys/utsname.h>
 #include <openenclave/internal/syscall/unistd.h>
 #include <openenclave/internal/trace.h>
+//#include "syscall_t.h"
```
Comment (dthaler) [inline syscall/syscall.c:29]: no commented out code

**Review by dthaler — CHANGES_REQUESTED** [review]
Please add test cases to the CI/CD.  (E.g., I think some of my comments are due to what looks like bugs, which would be easily caught by tests.)

**Comment by radhikaj** [comment]
Closing this for now. Please reopen when we resume working on it @anakrish

**Pull request closed without merging**